simulator/export/gdb/gdb_wrapper.cpp

```diff
@@ -39,12 +40,25 @@ catch (...) {
     return false;
 }
 
-bool GDBSim::create_inferior( Addr start_addr) const
+bool GDBSim::create_inferior( Addr start_addr, const char* const* argv, const char* const* envp) const try
 {
+    Addr sp = cpu->read_gdb_register( 29);
+    sp += ArgvLoader<int, Endian::native>( argv, envp).load_to( memory, sp);
```

Comment:
`Simulator` instances has method ```sizeof_register()```, you can use it to choose between uint32/uint64 (`int` is never an option).